src/cmap/connection_pool.ts

```diff
@@ -320,7 +320,7 @@ export class ConnectionPool extends TypedEventEmitter<ConnectionPoolEvents> {
           clearTimeout(waitQueueMember.timer);
         }
         if (!waitQueueMember[kCancelled]) {
-          waitQueueMember.callback(new MongoDriverError('connection pool closed'));
+          waitQueueMember.callback(new MongoResourceClosedError('Connection pool closed'));
```

Comment:
Should we just go ahead and create a `MongoConnectionPoolClosedError`? It feels somewhat out of place having the more generic form of this error directly instantiated.

Reply:
Sounds good! In general, I think it might be worth to make specific ClosedErrors for things that users will commonly run into (servers, streams, etc) and leaving `MongoResourceClosedError` as the general use case.

Reply:
In that case, this might be worth discussing with the team to get a better idea of the more common cases